lunabot_embedded/firmware/teensy_main/teensy_main.ino

```diff
@@ -22,50 +23,49 @@ void recv() {
 
   /* Now we are ready to decode the message. */
   pb_decode(&stream, RobotEffort_fields, &effort);
+  
+  //TODO, add timer if robot effort not changing for too long, exit?
+  KillSwitchRelay::logic(effort);
 
```

Comment:
Maybe put this in the main loop? for the weird case we lose comms with the jetson, kill switch should still be running since this function only runs when recv buffer has values

Reply:
To do this, we should move 
```
  pb_istream_t stream = pb_istream_from_buffer(buffer, sizeof(buffer));

  /* Now we are ready to decode the message. */
  pb_decode(&stream, RobotEffort_fields, &effort);
```
To before recv() is called (but still in the `if recv() > 0` block). then have the killswitch::logic and robot.cpp callbacks outside the if statement (but **make sure that effort is always zero-initialized** at the start of the loop to avoid unexpected actions)

thoughts?

Reply:
ALSO important. if you do this. you need to wrap the code with an if statement to fix the control frequency similar to TX:

```
  if (ms_until_send > TX_PERIOD) {
    ms_until_send -= TX_PERIOD;
    send();
    n = RawHID.send(buffer, 0);
  }
```

Reply:
Effort is zero intialized at the start of the code. Do you want to do it at the start of every loop? That seems like it would cause unexpected problems.

Reply:
you're right actually, don't zero initialize.